_docs-sources/iac/usage/composing-your-own-service.md

```diff
@@ -0,0 +1,193 @@
+
+To keep the configuration simple for this guide, we define a single route — `ANY /{proxy+}`. This tells the API Gateway to send any requests matching the path `/*` to the Lambda. This is an effective approach if you are using an API framework in the Lambda function code that can handle request routing. We will also set some defaults for the Lambda to not run in a VPC, have a maximum run time of 30 seconds, and 128MB of memory.
+
+```hcl title=gw_service_guide/serverless-api/main.tf
+module "lambda" {
+  source = "git::git@github.com:gruntwork-io/terraform-aws-lambda.git//modules/lambda?ref=v0.21.9"
+  name        = "${var.name}-name"
+  runtime     = var.lambda_runtime
+  source_path = var.lambda_source_path
+  handler     = var.lambda_handler
+  run_in_vpc  = false
+  timeout     = 30
+  memory_size = 128
+}
+
+module "api" {
+  source = "git::git@github.com:gruntwork-io/terraform-aws-lambda.git//modules/lambda-http-api-gateway?ref=v0.21.9"
+
+  name = "${var.name}-api-gw"
+  route_config = {
+    "ANY /{proxy+}": {
+        lambda_function_arn = module.lambda.function_arn
+    }
+  }
+}
+```
+
+### Specify the variables
+
+Now that you’ve defined the resources you want to create, you need to define the variables that you want to allow users to pass into the module. You can reference these values in the module using the var syntax, as visible in `gw_service_guide/serverless-api/main.tf`.
+
+```hcl title=gw_service_guide/serverless-api/variables.tf
+variable "name" {
+  description = "The name used to namespace all the resources, including the API Gateway and Lambda functions."
+  type        = string
+}
+
+variable "lambda_runtime" {
+  type        = string
+  description = "The runtime of the Lambda. Options include go, python, ruby, etc."
+}
+
+variable "lambda_source_path" {
+  type        = string
+  description = "The path to the directory containing the source to be deployed to lambda"
+}
+
+variable "lambda_handler" {
+  type        = string
+  description = "The name of the handler function that will be called as the entrypoint of the lambda"
+}
+```
+
+### Specify the outputs
+
+Next, define the outputs from the module. Outputs are convenient ways to pass values between modules when composing a service comprised of many modules. For this guide, we only want a single output — the URL for the API we are provisioning. For a module consumed by your company or team, you may want to define more outputs, to ensure that all resources created have associated outputs the consumers of your module can use as they see fit. Refer to the Library Reference for the [`Lambda function` module](../../reference/modules/terraform-aws-lambda/lambda/#reference) and [`HTTP API Gateway` module](../../reference/modules/terraform-aws-lambda/lambda-http-api-gateway/#reference) for a full list of outputs available.
```

Comment:
> For a module consumed by your company or team, you may want to define more outputs, to ensure that all resources created have associated outputs the consumers of your module can use as they see fit.

I had to read this sentence a few times, I think the second comma tripped me up. Can you re-word? maybe just: "You may want to define more outputs when developing a module for your company or team."

Reply:
Resolved in https://github.com/gruntwork-io/docs/pull/846/commits/822ebcde2f9c781a89b82f0786e8fa0da7bc9728